tests/ui/or-patterns/or-patterns-syntactic-fail.stderr

```diff
@@ -2,9 +2,15 @@ error: expected identifier, found `:`
   --> $DIR/or-patterns-syntactic-fail.rs:11:19
    |
 LL |     let _ = |A | B: E| ();
-   |                   ^ expected identifier
+   |             ----  ^ expected identifier
+   |             |
+   |             while parsing the body of this closure
    |
    = note: type ascription syntax has been removed, see issue #101728 <https://github.com/rust-lang/rust/issues/101728>
+help: you might have meant to open the body of the closure
+   |
+LL |     let _ = |A | { B: E| ();
+   |                  +
```

Comment:
This is a false positive

Reply:
I saw this, but I have no clue how we're meant to even start to try to parse such code in practice 😅

I feel it's fine, it still gives people some sense of what's happening.

Reply:
Yea... idk. I just feel like the suggestion doesn't help. But maybe it's better than without it?

Reply:
I can always special case it, but the previous error wasn't much better, tbqh